tools/djxl_ng_main.cc

```diff
@@ -386,22 +397,69 @@ int main(int argc, char** argv) {
   auto dec = JxlDecoderMake(/*memory_manager=*/nullptr);
   auto runner = JxlThreadParallelRunnerMake(
       /*memory_manager=*/nullptr, num_worker_threads);
+  std::vector<uint8_t> bytes;
   if (codec == jxl::extras::Codec::kJPG
 #if JPEGXL_ENABLE_JPEG
       && !FLAGS_pixels_to_jpeg
 #endif
   ) {
-    std::vector<uint8_t> jpeg_bytes;
+    std::vector<uint8_t> bytes;
     for (size_t i = 0; i < num_reps; ++i) {
-      if (DecompressJxlReconstructJPEG(compressed, jpeg_bytes, std::move(dec),
+      if (DecompressJxlReconstructJPEG(compressed, bytes, std::move(dec),
                                        std::move(runner)) != 0) {
         return EXIT_FAILURE;
       }
     }
-    if (WriteFile(filename_out, jpeg_bytes)) {
+    if (WriteFile(filename_out, bytes)) {
       return EXIT_FAILURE;
     };
     // TODO(firsching): handle non-reconstruct JPEG
+  } else if (codec == jxl::extras::Codec::kPNM) {
+    JxlDataType datatype = JXL_TYPE_UINT8;
```

Comment:
The format that contains this datatype is modified if the decoded data requires it, see line 283. There we change to `JXL_TYPE_UINT16` if it is necessary.

Reply:
Oh, yes, somehow this escaped my attention even as I looked for something along those lines.

What would you think of making the condition `bits_per_sample > 8` instead of `>= 16`?

Reply:
That would be needed if we want to accurately decode to e.g. 10-bit or 12-bit ppm...

Reply:
yeah sure, that would make sense!